examples/mac-example.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2020 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#include <mococrw/mac.h>
+#include <mococrw/util.h>
+#include <iostream>
+
+using namespace mococrw;
+
+std::unique_ptr<mococrw::HMAC> doHmac(const DigestTypes &digestType, const std::vector<uint8_t> &key,
```

Comment:
```suggestion
std::unique_ptr<mococrw::HMAC> doHmac(DigestTypes digestType, const std::vector<uint8_t> &key,
```

We don't usually pass simple types by reference, and IIRC `DigestTypes` is just an enum. The same applies to verify() and possibly others.